pkg/collector/parser/receiver_test.go

```diff
@@ -98,6 +98,19 @@ func TestReceiverFailsWhenPortIsntString(t *testing.T) {
 	assert.Nil(t, p)
 }
 
+func TestReceiverFailsWhenNameIsKubeletStats(t *testing.T) {
```

Comment:
I would do something like this with more comments on the test case and with a `endpoint` field in the test data that usually comes along with `kubeletstats` receiver. Also, will add additional checks like

```
func TestIgnorekubeletstatsEndpoint(t *testing.T) {
	// ignore "kubeletstats" receiver endpoint field, this is special case
	// as this receiver gets parsed by generic receiver parser
	builder := parser.NewGenericReceiverParser(logger, "kubeletstats", map[interface{}]interface{}{
		"endpoint": "0.0.0.0:9000",
	})

	// test
	ports, err := builder.Ports()

	// verify
	assert.NoError(t, err)
	assert.Len(t, ports, 0)
}
```

Reply:
Hi @VineethReddy02, thanks a lot for all the help and this learning, have tried to modify the code according to suggestions, kindly review it at your convenient time, thanks a lot!